addons/mail/models/mail_channel_partner.py

```diff
@@ -45,3 +45,18 @@ def write(self, vals):
             if {'channel_id', 'partner_id', 'partner_email'} & set(vals):
                 raise AccessError(_('You can not write on this field'))
         return super(ChannelPartner, self).write(vals)
+
+    def mail_channel_partner_format(self, partner_infos=None):
+        members = self.read(['id', 'partner_id'])
+        partner_infos = partner_infos or {}
+        formatted_members = []
+        for member in members:
+            partner_data = partner_infos.get(member['partner_id'][0], {
+                'id': member['partner_id'][0],
+                'display_name': member['partner_id'][1]
+            })
+            formatted_members.append({
+                'id': member['id'],
+                'partner': partner_data,
+            })
+        return formatted_members
```

Comment:
double check all of this, it looks like a very complicated way to write something that is very simple